src/kvstore.h

```diff
@@ -108,6 +110,21 @@ void kvstoreDictSetAtLink(kvstore *kvs, int didx, void *kv, dictEntryLink *link,
 void kvstoreDictSetKey(kvstore *kvs, int didx, dictEntry* de, void *key);
 void kvstoreDictSetVal(kvstore *kvs, int didx, dictEntry *de, void *val);
 
+/**************** estore API (See comment in estore.c) ************************/
+estore *estoreCreate(EbucketsType *type, int num_buckets_bits);
```

Comment:
why put estore in kvstore but in a separatre file?

Reply:
I did it at start, but then their common logic, such as fenwick tree, i needed to move as well. So since they have a lot of in common and their API is similar, i ended up having both in the same file.

Reply:
IMHO, kvstore and estore share many similarities. As long as there is no additional need for the Fenwick tree outside, i feel comfortable to leave it as is. My main concern, however, lies with the other large files. In my view, our priority should be to modernize the build system, move away from Makefile, introduce native folder support, and work toward open/close APIs. Once that foundation is in place, with proper unit testing, we should gradually decompose the large files into modular components.

Reply:
i don't feel strongly about it, with the way things currently are, we have flexibility of putting two infrastructures in the same file if we feel that's nicer.
i do tend to agree with the 3 points above, specifically because of the use of server.h and the fact the fenwick was refactored anyway.